File: src/commands/PrefixCommand.js
const Command = require('../structures/Command')

class PrefixCommand extends Command {
  constructor () {
    super('prefix')
    this.name = 'Prefix'
    this.category = 'server'
    this.memberPermissions = ['MANAGE_GUILD']
    this.usage = 'prefix'
  }

  async run (message, args, server, { t }) {
    if (!args[0]) return this.explain(message)
    if (args[0].split('').length >= 4) return message.reply(t('commands:prefix.maxLength'))
    server.prefix = args[0]
    server.save().then(message.reply(t('commands:prefix.changed', { prefix: args[0] })))
  }
}

module.exports = PrefixCommand

File: src/main/kotlin/com/mystery/meteora/client/commands/PrefixCommand.kt
package com.mystery.meteora.client.commands

import com.mystery.meteora.controller.Config
import com.mystery.meteora.controller.Helper
import com.mystery.meteora.controller.model.Guilds
import com.mystery.meteora.controller.translate
import com.mystery.meteora.handler.annotations.Command
import com.mystery.meteora.handler.annotations.Description
import com.mystery.meteora.handler.annotations.Module
import com.mystery.meteora.handler.annotations.Usage
import com.mystery.meteora.handler.modules.BaseModule
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import org.litote.kmongo.*

@Module("Prefix", "management")
@Description("prefix.description")
@Usage("prefix.usage")

class PrefixCommand(ctx: MessageReceivedEvent, args: String, prefix: String, config: Config) : BaseModule(ctx, args, prefix, config) {
  @Command("prefix")
  fun prefix() {
    val permission = context.member?.hasPermission(Permission.MESSAGE_MANAGE)
    if (!permission!!) {
      context.channel.sendMessage("prefix.insufficientPerms".translate(config, context.guild.id)).queue()
      return
    }
    if (args == "") {
      Helper().explain(context, "prefix", "Prefix", prefix, config!!)
      return
    }
    val argsArray = args.split(' ').toTypedArray()
    val prefix = argsArray[0]
    if (prefix.length > 3) {
      context.channel.sendMessage("prefix.maxValue".translate(config, context.guild.id)).queue()
    } else {
      val client = KMongo.createClient(config!!.config?.databaseConfig?.connectionUri!!)
      val database = client.getDatabase("meteora")
      val collection = database.getCollection("guilds")
      collection.updateOne(Guilds::_id eq context.guild.id, Guilds::prefix setTo prefix)
      context.channel.sendMessage("prefix.changed".translate(config, context.guild.id, prefix)).queue()
      client.close()
    }
  }
}